diff --color -uNr a/client/options.c b/client/options.c
--- a/client/options.c	2022-10-05 15:57:12.661717853 +0200
+++ b/client/options.c	2022-10-05 15:56:44.854549898 +0200
@@ -729,9 +729,9 @@
 
 	if (!print_stdin) {
 		if (progname) {
-			printf("usage: %s [args]\n", progname);
+			printf("usage: %s [args]\n\nNOTE: reboot-action does not power on nodes that are powered off.\n\n", progname);
 		} else {
-			printf("usage: fence_virt [args]\n");
+			printf("usage: fence_virt [args]\n\nNOTE: reboot-action does not power on nodes that are powered off.\n\n");
 		}
 	}
 
@@ -761,7 +761,8 @@
 	printf("<?xml version=\"1.0\" ?>\n");
 	printf("<resource-agent name=\"%s\" shortdesc=\"Fence agent for virtual machines\">\n", basename(progname));
 	printf("<longdesc>%s is an I/O Fencing agent which can be used with "
-	       "virtual machines.</longdesc>\n", basename(progname));
+	       "virtual machines.\n\nNOTE: reboot-action does not power on nodes that are powered off."
+	       "</longdesc>\n", basename(progname));
 	printf("<parameters>\n");
 
 	for (x = 0; x < strlen(optstr); x++) {
diff --color -uNr a/man/fence_virt.8 b/man/fence_virt.8
--- a/man/fence_virt.8	2019-11-14 16:29:26.000000000 +0100
+++ b/man/fence_virt.8	2022-10-05 16:01:44.001358622 +0200
@@ -15,6 +15,7 @@
 For compatibility, fence_xvm may talk to fence_xvmd from linux-cluster
 release 2 or later.
 
+NOTE: reboot-action does not power on nodes that are powered off.
 .P
 fence_virt and fence_xvm accept options on the command line as well
 as from stdin.  The fencing daemon sends parameters through stdin
